gst/datarepo/gstdatareposink.c

```diff
@@ -719,7 +720,7 @@ gst_data_repo_sink_write_json_meta_file (GstDataRepoSink * sink)
   json_object_set_int_member (sink->json_object, "total_samples",
       sink->total_samples);
 
-  if (sink->is_flexible_tensors) {
+  if (sink->data_type == GST_DATA_REPO_DATA_TENSOR && !sink->is_static_tensors) {
```

Comment:
Instead of 
```c
if (cond) {
    return ret
}
```
Use 
```c
if (!cond) {
    return something;
}

Code for cond == TRUE is here.
```

Reply:
Thank. but
This sentence is an if and else.
if we change this as below. so it looks like the code doesn't need to be changed.
```
  if (sink->data_type != GST_DATA_REPO_DATA_TENSOR || sink->is_static_tensors) {
    json_object_set_int_member (sink->json_object, "sample_size",
        sink->sample_size);
  } else {
    json_object_set_array_member (sink->json_object, "sample_offset",
        sink->sample_offset_array);
    json_object_set_array_member (sink->json_object, "tensor_size",
        sink->tensor_size_array);
    json_object_set_array_member (sink->json_object, "tensor_count",
        sink->tensor_count_array);
  }
```